nebu/cli/get.py

```diff
@@ -210,21 +215,55 @@ def _write_node(node, base_url, out_dir, book_tree=False, get_resources=False,
                     file_resp = requests.get(url)
                     filepath.write_bytes(file_resp.content)
 
+        # Cache/store sha1-s for resources in a 'dot' file
+        gen_resources_sha1_cache(write_dir, resources)
+
         if pbar is not None:
             pbar.update(1)
 
+    elif book_tree:  # No metadata to fetch, is a subcollection
+        # Need to create subcol dir
+        out_dir = gen_tree_path(ctx, out_dir, node, None, lvl, pos)
+        out_dir.mkdir()
+
     if 'contents' in node:  # Top-level or subcollection - recurse
         lvl += 1
         if lvl not in pos:
             pos[lvl] = 0
         if lvl == depth:  # Reset counter for bottom-most level: pages
             pos[lvl] = 0
         for child in node['contents']:
-            #  HACK - the silly don't number Preface/Introduction logic
-            if ((lvl == 1 and pos[1] == 0 and 'Preface' in child['title']) or
-                    (pos[lvl] == 0 and child['title'] == 'Introduction')):
-                pos[lvl] = 0
-            else:
-                pos[lvl] += 1
-            _write_node(child, base_url, out_dir, book_tree, get_resources,
+            pos[lvl] += 1
+            _write_node(ctx, child, base_url, out_dir,
+                        book_tree, get_resources,
                         pbar, depth, pos, lvl)
```

Comment:
This is all still pretty convoluted, but I hope factoring the tree-numbering decision out helps make the code flow a bit more logical. It's still too much sequential code (hard to see that this is a recursive function, basically) So, future refactor target, for sure.